crates/subspace-service/src/dsn.rs

```diff
@@ -119,8 +94,7 @@ where
             .unwrap_or(Ok(StubNetworkingParametersManager.boxed()))?
     };
 
-    let provider_storage =
-        NodeProviderStorage::new(peer_id, piece_cache.clone(), external_provider_storage);
+    let provider_storage = piece_cache.clone();
```

Comment:
This line is not necessary anymore, just replace `provider_storage.clone(),` with `piece_cache.clone()` below. Surprised Clippy didn't catch it.